pkg/provision/engine.go

```diff
@@ -182,6 +182,19 @@ func (e *Engine) provision(ctx context.Context, r *Reservation) error {
 		return errors.Wrapf(err, "failed to cache reservation %s locally", r.ID)
 	}
 
+	// If an update occurs on the network we don't increment the counter
```

Comment:
I don't believe you need this, you can just check the cache `e.cache` if this reservation exists (and still active) or not (before adding it to the cache) then only increment if didn't exist before.